packages/x-charts/src/RadarChart/seriesConfig/seriesProcessor.ts

```diff
@@ -0,0 +1,29 @@
+import { type SeriesProcessor } from '../../internals/plugins/models/seriesConfig';
+import type { DefaultizedRadarSeriesType } from '../../models';
+import type { SeriesId } from '../../models/seriesType/common';
+
+const radarValueFormatter = ((v) =>
+  v == null ? '' : v.toLocaleString()) as DefaultizedRadarSeriesType['valueFormatter'];
+
+const seriesProcessor: SeriesProcessor<'radar'> = (params, _) => {
+  const { seriesOrder, series: seriesMap } = params;
+
+  const completedSeries: Record<SeriesId, DefaultizedRadarSeriesType> = {};
+
+  seriesOrder.forEach((seriesId) => {
+    const series = seriesMap[seriesId];
+
+    completedSeries[seriesId] = {
+      labelMarkType: 'square',
+      valueFormatter: radarValueFormatter,
+      ...series,
```

Comment:
I think we should start using this pattern instead:

```suggestion
      ...series,
      valueFormatter: series.valueFormatter ?? radarValueFormatter,
```

`series.valueFormatter` can be set to `undefined`, in which case the previous solution wouldn't work. Proof:

<img width="229" height="176" alt="Image" src="https://github.com/user-attachments/assets/e2ba150f-0de0-4570-96fe-fba01cd15aff" />

We can set `valueFormatter` in a `RadarChart` to `undefined` without a type error, so this is a bug that we have across several places in our codebase.